src/Gui/DlgMaterialPropertiesImp.cpp

```diff
@@ -78,31 +84,65 @@ void DlgMaterialPropertiesImp::setupConnections()
             this, &DlgMaterialPropertiesImp::onSpecularColorChanged);
     connect(ui->shininess, qOverload<int>(&QSpinBox::valueChanged),
             this, &DlgMaterialPropertiesImp::onShininessValueChanged);
+    connect(ui->buttonReset, &QPushButton::clicked,
+            this, &DlgMaterialPropertiesImp::onButtonReset);
+    connect(ui->buttonDefault, &QPushButton::clicked,
+            this, &DlgMaterialPropertiesImp::onButtonDefault);
 }
 
 QColor DlgMaterialPropertiesImp::diffuseColor() const
 {
     return ui->diffuseColor->color();
 }
 
+App::Material DlgMaterialPropertiesImp::getMaterial()
+{
+    for (std::vector<ViewProvider*>::iterator it = Objects.begin(); it != Objects.end(); ++it) {
+        App::Property* prop = (*it)->getPropertyByName(material.c_str());
+        if (prop && prop->isDerivedFrom<App::PropertyMaterialList>()) {
+            auto ShapeAppearance = static_cast<App::PropertyMaterialList*>(prop);
+            auto& materials = ShapeAppearance->getValues();
+            auto mat = materials[0];
+
+            mat.setType(App::Material::USER_DEFINED);
+
+            // Reset any texture data
+            mat.image = "";
+            mat.imagePath = "";
+            mat.uuid = "";
+
+            return mat;
+        }
+    }
+
+    return App::Material::getDefaultAppearance();
+}
+
+App::Color DlgMaterialPropertiesImp::getColor(const QColor& color) const
+{
+    float r = (float)color.red() / 255.0F;
+    float g = (float)color.green() / 255.0F;
+    float b = (float)color.blue() / 255.0F;
+
+    return App::Color(r, g, b);
+}
+
 /**
  * Sets the ambient color.
  */
 void DlgMaterialPropertiesImp::onAmbientColorChanged()
 {
-    QColor col = ui->ambientColor->color();
-    float r = (float)col.red() / 255.0f;
-    float g = (float)col.green() / 255.0f;
-    float b = (float)col.blue() / 255.0f;
-    App::Color ambient(r, g, b);
+    if (updateColors) {
+        App::Color ambient = getColor(ui->ambientColor->color());
 
-    for (std::vector<ViewProvider*>::iterator it= Objects.begin(); it != Objects.end(); ++it) {
-        App::Property* prop = (*it)->getPropertyByName(material.c_str());
-        if (prop && prop->isDerivedFrom<App::PropertyMaterial>()) {
-            auto ShapeMaterial = static_cast<App::PropertyMaterial*>(prop);
-            App::Material mat = ShapeMaterial->getValue();
-            mat.ambientColor = ambient;
-            ShapeMaterial->setValue(mat);
+        for (std::vector<ViewProvider*>::iterator it= Objects.begin(); it != Objects.end(); ++it) {
+            App::Property* prop = (*it)->getPropertyByName(material.c_str());
+            if (prop && prop->isDerivedFrom<App::PropertyMaterialList>()) {
+                auto ShapeAppearance = static_cast<App::PropertyMaterialList*>(prop);
+                auto mat = getMaterial();
+                mat.ambientColor = ambient;
+                ShapeAppearance->setValue(mat);
+            }

```

Comment:
This is a lot of boilerplate repeated a half dozen times with only a single line change -- in the future it would be a good idea to refactor to eliminate that code duplication (and to modernize the for loop).

Reply:
Yes. This is in progress as I work on textures. My thinking was make it work first